doc/panama_ffi.md

```diff
@@ -12,75 +12,35 @@ Before we dive into the specifics of the foreign function support, it would be u
 
 ```java
-MemorySegment raw = ... //obtain address from native code
-try (Arena arena = Arena.openConfined()) {
-    MemorySegment segment = MemorySegment.ofAddress(raw.address(), 100, arena.scope());
-    int x = segment.get(JAVA_INT, 0);
-}
+MemorySegment foreign = someSegment.get(ValueLayout.ADDRESS, 0); // size = 0
+                                   .reinterpret(4)               // size = 4
+int x = foreign.get(ValueLayout.JAVA_INT, 0);                    // ok
 ```
 
-Alternatively, clients can obtain an *unbounded* address value layout. This is done using the `ValueLayout.OfAddress::asUnbounded` method (which is also a restricted method). When an access operation uses an unbounded address value layouts, the runtime will wrap any corresponding raw addresses with native segments with <em>maximal</em> size (i.e. `Long.MAX_VALUE`). As such, these segments can be accessed directly, as follows:
+In some cases, a client might additionally want to assign new temporal bounds to a zero-length memory segment. This can be done using another variant of the `MemorySegment::reinterpret` method, which returns a new native segment with the desired size and temporal bounds:
 
 ```java
-MemorySegment foreign = someSegment.get(ValueLayout.ADDRESS.asUnbounded(), 0); // wrap address into segment (size = Long.MAX_VALUE)
-int x = foreign.get(ValueLayout.JAVA_INT, 0); //ok
-```
-
-Which approach is taken largely depends on the information that a client has available when obtaining a memory segment wrapping a native pointer. For instance, if such pointer points to a C struct, the client might prefer to resize the segment unsafely, to match the size of the struct (so that out-of-bounds access will be detected by the API). In other instances, however, there will be no, or little information as to what spatial and/or temporal bounds should be associated with a given native pointer. In these cases using an unbounded address layout might be preferable.
-### Segment allocators
-
-Idiomatic C code implicitly relies on stack allocation to allow for concise variable declarations; consider this example:
-
-```c
-int arr[] = { 0, 1, 2, 3, 4 };
-```
-
-A variable initializer such as the one above can be implemented as follows, using the Foreign Memory Access API:
-
-```java
-try (Arena arena = Arena openConfined()) {
-    MemorySegment arr = MemorySegment.allocateNative(MemoryLayout.sequenceLayout(5, JAVA_INT), arena.scope());
-    for (int i = 0 ; i < 5 ; i++) {
-        arr.setAtIndex(JAVA_INT, i, i);
-    }
+MemorySegment foreign = null;
+try (Arena arena = Arena.ofConfined()) {
+      foreign = someSegment.get(ValueLayout.ADDRESS, 0)           // size = 0, scope = always alive
+                           .reinterpret(4, arena.scope(), null);  // size = 4, scope = arena.scope()
```

Comment:
This should pass the arena now, I think.
```suggestion
                           .reinterpret(4, arena, null);  // size = 4, scope = arena.scope()
```

Reply:
Good catch - I'll re-check. Main issue is that the edits were done when the method accepted a scope. Perhaps some other editing is required.